### PR TITLE
clutter: Remove docs from build

### PR DESCRIPTION
**Pull request by A6GibKm**
Fixes: https://github.com/flathub/shared-modules/issues/193

### CODE REVIEW

File: clutter/clutter.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
     "name": "clutter-gtk",
+    "cleanup": [
+        "/share/gtk-doc"
```
Comment (wjt) [inline clutter/clutter.json:4]: This is insufficient, you need to clean up the submodules too.

**Comment by wjt** [comment]
```
diff --git a/clutter/clutter.json b/clutter/clutter.json
index 6795c98..f7b9b04 100644
--- a/clutter/clutter.json
+++ b/clutter/clutter.json
@@ -13,6 +13,9 @@
     "modules": [
         {
             "name": "cogl",
+            "cleanup": [
+                "/share/gtk-doc"
+            ],
             "config-opts": [
                 "--disable-cogl-gst",
                 "--disable-gtk-doc",
@@ -29,6 +32,9 @@
         },
         {
             "name": "clutter",
+            "cleanup": [
+                "/share/gtk-doc"
+            ],
             "config-opts": [
                 "--disable-gtk-doc",
                 "--enable-egl-backend",
@@ -44,6 +50,9 @@
         },
         {
             "name": "clutter-gst",
+            "cleanup": [
+                "/share/gtk-doc"
+            ],
             "config-opts": [
                 "--disable-gtk-doc"
             ],
```

**Comment by A6GibKm** [comment]
done

**Comment by wjt** [comment]
Looks good. If you're feeling keen you could also add:

```patch
diff --git a/clutter/clutter.json b/clutter/clutter.json
index 0f0967a..a47b207 100644
--- a/clutter/clutter.json
+++ b/clutter/clutter.json
@@ -20,7 +20,8 @@
                 "--enable-wayland-egl-platform"
             ],
             "cleanup": [
-                "/share/gtk-doc"
+                "/share/gtk-doc",
+                "/share/cogl/examples-data"
             ],
             "sources": [
                 {
```

There is some other stuff that could be cleaned up, like some valgrind suppressions and all of `/include`, but this has knocked 8MB off. Of the remaining 13M the biggest piece is the gobject-introspection stuff, which the app might need.

**Comment by A6GibKm** [comment]
Thanks, I removed a cleanup for the examples.

The introspection stuff is massive, I wish we had a way to pass arguments  like `--enable-instrospection=false` to shared modules.

**Comment by hfiguiere** [comment]
> Thanks, I removed a cleanup for the examples.
> 
> The introspection stuff is massive, I wish we had a way to pass arguments like `--enable-instrospection=false` to shared modules.

`--disable-introspection` doesn't work?

**Comment by A6GibKm** [comment]
> > Thanks, I removed a cleanup for the examples.
> > The introspection stuff is massive, I wish we had a way to pass arguments like `--enable-instrospection=false` to shared modules.
> 
> `--disable-introspection` doesn't work?

I don't see why not, but then we either have two provide two versions of the module or we left out apps that use introspection data.

**Pull request merged**